blocks/header/header.js

```diff
@@ -85,6 +89,107 @@ function toggleMenu(nav, navSections, forceExpanded = null) {
   }
 }
 
+async function fetchSearchData() {
+  if (!window.searchData) {
+    const resp = await fetch(`/${window.location.pathname.split('/')[1]}/query-index.json`);
+    const json = await resp.json();
+    window.searchData = json;
+  }
+  return window.searchData;
+}
+
+async function search(value) {
+  const searchData = await fetchSearchData();
+  const hits = [];
+  for (let i = 0; i < searchData.data.length; i += 1) {
+    const e = searchData.data[i];
+    const text = [e.title, e.description].join(' ').toLowerCase();
+    if (text.includes(value.toLowerCase())) {
+      hits.push(e);
+    }
+  }
+  return hits;
+}
+
+function decorateSearch(block) {
+  const searchSection = block.querySelector('a > .icon-search');
+  if (searchSection) {
+    const searchElement = document.createElement('div');
+    searchElement.classList.add('nav-search');
+    searchSection.parentElement.parentElement
+      .replaceChild(searchElement, searchSection.parentElement);
+    searchElement.appendChild(searchSection);
+
+    const aside = document.createElement('aside');
+    aside.classList.add('nav-search-aside');
+    aside.classList.add('nav-search-aside-empty');
+
+    const input = document.createElement('input');
+    input.classList.add('nav-search-input');
+    input.type = 'search';
+    input.value = new URL(window.location).searchParams.get('ee_search_query');
+
+    input.addEventListener('input', async (e) => {
+      const url = new URL(window.location);
+      if (e.target.value.length >= 3) {
+        url.searchParams.set('ee_search_query', e.target.value);
+        const hits = await search(e.target.value);
+        if (hits.length > 0) {
+          aside.classList.remove('nav-search-aside-empty');
+          aside.innerHTML = `<h1 class="nav-search-result-title">Search Results for: ${input.value}</h1>
+          <div class="nav-search-result-title-divider"><span class="nav-search-result-title-divider-separator"/></div>`;
+        } else {
+          aside.classList.add('nav-search-aside-empty');
+          aside.innerHTML = '';
+        }
+        hits.forEach((hit) => {
+          const wrapper = document.createElement('div');
+          wrapper.classList.add('nav-search-wrapper');
+          wrapper.innerHTML = `<h3 class='nav-search-title'><a href='${hit.path}'>${hit.title}</a></h3>
+          <div class='nav-search-description'>${hit.description}</div>`;
+          aside.appendChild(wrapper);
+        });
+      } else {
+        url.searchParams.delete('ee_search_query');
+        aside.innerHTML = '';
+        aside.classList.add('nav-search-aside-empty');
+      }
+      // eslint-disable-next-line no-restricted-globals
+      history.replaceState(null, '', url);
+    });
+
+    let active = input.value;
+
+    if (active) {
+      searchElement.prepend(input);
+      searchElement.append(aside);
+      input.dispatchEvent(new Event('input', { bubbles: true }));
+    }
+
+    searchSection.addEventListener('click', (event) => {
```

Comment:
for readability, i would extract the listener function to its own named function and just reference it here.

Reply:
Done